week-1-ts-ecosystem/w01e03-object-oriented-structure/src/models/buyNowProduct.ts

```diff
@@ -0,0 +1,7 @@
+import { BaseProduct } from "./baseProduct";
+
+export class BuyNowProduct extends BaseProduct {
```

Comment:
Do tego podszedłeś bardzo dobrze, ale i tutaj widzę "room for improvements"

Jeśli złapiesz / zaakceptujesz tą koncepcję `abstract` to zobacz
że tutaj pole `type: "Buy Now"` jest o tyle bez sensu, że teraz w każdej instancji musisz podawać jeden nadmiarowy argument, który jest zbędny bo to klasa określa teraz że to `BuyNowProdut`

jeśli chcesz koniecznie to pole zostawić to z `abstract` wyżej wymusi jego zdefiniowanie np.:

```typescript
export class BuyNowProduct extends BaseProduct {

  public type = "Buy Now"

//....
}

```

Ale IMHO - ten typ to teraz dodatkowy koszt a nie ułatwienie.

Zawsze możemy poznać co to za produkt. Wystarczy sprawdzić hierarchię klas
```typescript
const item = new BuyNowProduct (/*....*/)

console.log(item instanceof BaseProduct) // true
// ok to jest product

console.log(item instanceof BuyNowProduct) // true
// ok to jest BuyNowProduct
```

Reply:
Podobnie w pozostałych klasach.